lib/elixir/lib/gen_event.ex

```diff
@@ -783,7 +784,7 @@ defmodule GenEvent do
   end
 
   @doc false
-  def system_code_change([name, handlers, hib], module, old_vsn, extra) do
+  def system_code_change([name, handlers, hib], module, old_vsn, extra) when is_atom(module) do
```

Comment:
Same, it is a callback. I would prefer to not have the guard because the module name is validated upfront elsewhere.